src/components/CommunityHolderGrid/CommunityHolderGridItem.tsx

```diff
@@ -0,0 +1,116 @@
+import InteractiveLink from 'components/core/InteractiveLink/InteractiveLink';
+import { VStack } from 'components/core/Spacer/Stack';
+import { BaseM } from 'components/core/Text/Text';
+import { useReportError } from 'contexts/errorReporting/ErrorReportingContext';
+import { CouldNotRenderNftError } from 'errors/CouldNotRenderNftError';
+import Link from 'next/link';
+import { useMemo } from 'react';
+import { graphql, useFragment } from 'react-relay';
+import styled from 'styled-components';
+import { getOpenseaExternalUrl } from 'utils/getOpenseaExternalUrl';
+import getVideoOrImageUrlForNftPreview from 'utils/graphql/getVideoOrImageUrlForNftPreview';
+import { graphqlTruncateUniversalUsername } from 'utils/wallet';
+import { CommunityHolderGridItemFragment$key } from '__generated__/CommunityHolderGridItemFragment.graphql';
+
+type Props = {
+  holderRef: CommunityHolderGridItemFragment$key;
+};
+
+export default function CommunityHolderGridItem({ holderRef }: Props) {
+  const token = useFragment(
+    graphql`
+      fragment CommunityHolderGridItemFragment on Token {
+        dbid
+        name
+        tokenId
+        contract {
+          contractAddress {
+            address
+          }
+        }
+        media {
+          __typename
+
+          ... on ImageMedia {
+            previewURLs @required(action: NONE) {
+              large @required(action: NONE)
+            }
+          }
+
+          ... on UnknownMedia {
+            previewURLs @required(action: NONE) {
+              large @required(action: NONE)
+            }
+          }
+        }
+        owner {
+          username
+          universal
+
+          ...walletTruncateUniversalUsernameFragment
+        }
+
+        ...getVideoOrImageUrlForNftPreviewFragment
+      }
+    `,
+    holderRef
+  );
+  const { tokenId, contract, owner } = token;
+
+  const usernameWithFallback = owner ? graphqlTruncateUniversalUsername(owner) : null;
+
+  const profileLink = owner?.universal
+    ? `https://opensea.io/${owner?.username}`
+    : `/${owner?.username}`;
+
+  const reportError = useReportError();
+  const previewUrlSet = getVideoOrImageUrlForNftPreview(token, reportError);
+
+  if (!previewUrlSet?.urls.large) {
+    throw new CouldNotRenderNftError('CommunityHolderGridItem', 'could not find large image url');
+  }
+
+  const openSeaExternalUrl = useMemo(() => {
+    if (contract?.contractAddress?.address && tokenId) {
+      return getOpenseaExternalUrl(contract.contractAddress.address, tokenId);
+    }
+
+    return null;
+  }, [contract?.contractAddress?.address, tokenId]);
+
+  return (
+    <VStack gap={8}>
+      <Link href={openSeaExternalUrl || ''} passHref>
```

Comment:
really great question @jakzaizzat! a few ideas:

**short term:** we rename the current `NftDetailView` into `CollectionTokenDetailView` and create a new file/directory called `TokenDetailView`. this will look almost identical to the current `NftDetailView`, except the fragment will work off the `Token` and ignore all features related to CollectionTokens.

**long term:** we have the backend return `CollectionToken`. the problem here is that non-gallery users' tokens are not part of any given collection; in fact, some tokens may belong to gallery users but not be displayed on their profile, which means the data type that's returned from the `tokensInCommunity` query may be inconsistent.

for now i vote that we roll the short-term solution.

Reply:
Thanks a lot robin! Let's do the short term solution